### ISSUE TITLE
Unit conversion in activities/calculation setups

### ISSUE THREAD

**Issue by marc-vdm** [LCA-ActivityBrowser/activity-browser#1132]
### Feature request

I want to suggest adding the following big feature:

An activity's exchanges are fundamentally just dimensionless numbers, the units we attach to them are metadata. 
Practitioners sometimes work with data in other units than are standard supplied by their background databases (e.g. ecoinvent).

I want to suggest a 'unit conversion' layer on top of Brightway _exchanges_. We can store the original amounts and units of activities, but we can store an additional attribute (e.g. `AB_unit`) in the exchange with a unit conversion (e.g. `{'kWh>MJ': 3.6}`), which would be shown to users in AB instead of the original amount and unit.

This would mean that if a user has an exchange of 1kWh of electricity, they can choose MJ (e.g. from a dropdown) as unit and AB would instead show 3.6MJ. For this, all we would need to do is store MJ as new unit.

What I propose is that all unit fields are converted from `str` fields to dropdowns/menus of some sort (though we essentially still store a `str`)

### Relevant things to consider:

__Unit conversions:__
For unit conversions we could use a standard library; [QuantiPhy](https://quantiphy.readthedocs.io/en/stable/) looks good to me (quite some features, widely used, actively maintained, well documented), but there may be others that could also work. We would probably need to develop something on top of the library (I think just a `dict` that we store on disk should suffice) where we can store custom units and custom conversion factors (e.g. `kg of oil` to  `m3 of oil` is different for different oils etc.). Additionally; users may be using more abstract units like product lifetimes in years etc, which should also be supported.

We should keep in mind that we need to manage 2 things here:
- Units: a measure of something
- Unit conversions: The scaling amount to go from one unit to another

__Unit manager UI:__
We need to create a custom UI part for the units manager somewhere to take care of conversions. We should decide whether this would be a tab or a dialog/wizard.

__Activity details: Product:__
When editing an activity, the user can edit a product unit. When this unit is changed in the same category (e.g. `weight` >> `kg` to `lb` or `tonne`). AB could give a popup offering to update the Exchanges to scale them to the same unit.
Here we should figure out the following potential challenges:
- What if user changes a to another category? (e.g. `weight`, `kg` >> `volume`, `m3`)
  - This could potentially be fixed by offering custom conversion factors. 
- Should this re-scale all input exchanges to the new unit, or keep the old inputs?
- Should this re-scale all 'downstream consumers' to the new unit, or keep the old one? 

__Activity details: Exchanges:__
When editing an activity, the user can choose relevant unit conversions (or create them manually) from a dropdown. The amount shown in the field would be automatically updated to the new unit (so it is still the same in the original amount). This should be possible for both Technosphere and Biosphere exchanges.
Here we should figure out the following potential challenges:
- If a user chooses a unknown conversion (e.g. `kg` >> `MJ` -which could be a real conversion, if they are modeling the burning of wood for heat for example-, user should be prompted to create a new custom conversion factor between the units) These conversion factors should be able to be stored either with product attached, or not (`kg` >> `m3` is different for water, oil, natural gas etc`)

__LCA setup and results__
Here we could also allow unit conversion, e.g. when you want to compare something from two different databases, and one is in `kg` while the other is in `tonnes`. User could then choose to scale per reference flow in the calculation setup. This conversion should be stored somewhere relevant for the setup, not changing the actual activities. All results should then also be shown with the 'new' unit.

__Unit naming conventions:__
Another thing to consider is that not all databases (and user created databases) will use the same unit naming conventions for the same units, e.g. `kg` vs `kilogram`. We should probably offer standardized unit naming for users when they create new activities, though they should be able to create custom ones too. We could do this by letting users choose units from a dropdown list (with an option `create new`) instead of just writing strings for units. I'm not sure if it's possible UI-wise, but it may be nice to have sub-menus in the dropdown (e.g. `weight` >> `kg`, `tonne` .., `volume` >> `liter`, `m3` ..) This dropdown would be great for any place users choose units.

__Data per Activity vs per Exchange:__
Mostly a summary of above challenges: I think we should support standard changing of units (which already exists) on the Activity level (so for products). For exchanges however (inputs into other activities), I think we should allow the actual conversions with new units discussed above. However: When the production unit is changed of an activity, we should then ask user two things: 1) if they want to re-scale all amounts of the activity exchanges (e.g. if I scale `kg` to `tonne`, AB should offer to divide all inputs by 1000) and 2) if they want to change all amounts in the downstream consumers (e.g. if I scale `kg` to `tonne`, AB should offer to divide all downstream consumption amounts by 1000). AB should also consider extra challenges here. If an input exchange was already scaled and the unit is changed, AB should be able to deal with re-scaling or creating new conversion factors or something.

**Comment by marc-vdm** [comment]
Closing as waiting on
https://github.com/sentier-dev/units
